sections/design/why-spa.tex

```diff
@@ -0,0 +1,48 @@
+\section{Single Page Application or Multi Page Application}
+There are two types of applications that can be created.
+One is a Single Page Application (SPA), the other is a Multi Page application (MPA).
+In this section we will be looking into the two types of architectures and explore the advantages and disadvantages that they have.
+
+\subsection{SPAs}
+What an SPA allows you to do is build an application that simulates that of a desktop application. 
+When a new page is loaded only the necessary content of the previous page is updated.
+JavaScript is often used when developing SPAs but it is often used with some library or framework like React to make the development easier.
+There are also other frameworks such as Angular or Vue which architecture allow for development of dynamic web application just as React does. 
+\\\\
+The main advantages of an SPA is the high speed that comes the unnecessity to load all elements with each new page but only load the changed elements. 
+The frameworks used also makes fast development possible because of the powerful tools that are provided. 
+It is also easy to create a mobile app based on the finished code of an SPA \cite{SPAvsMPAMerehead}.
+SPAs also have greate caching possibilities and the local data can be used offline for when the user has problems with their connection. 
+The pages can then still be shown to the user even with connectivity issues and the local data will be synchronized once the connection is up again \cite{SPAvsMPARuby}.
+\\\\
+Disadvantages of an SPA is the poor optimization for search engines. Most of the pages will not be available to scan because of the way an SPA works. 
+Some users will also have JavaScript disabled in their browsers, and since the SPA is reliant on JavaScript to load new page content all the time, the pages will not be able to load correctly.
+Generally SPAs have less security but with use of the modern frameworks mentioned the security of the page is significantly improved.
+SPAs also can not take advantes of browser history since the back button will can not go to the previous page since there only is one. 
+There are solutions that can be used for this issue such as using a history API \cite{SPAvsMPAMerehead}.
+
+\subsection{MPAs}
+MPAs are different to SPAs in that they have a more standard architecture. When a new page is loaded it will send a request to the server and all the data will be updated even if it is just a small amount. A MPA will therefore spent a bit more time loading a new page because it has to update everything.
```

Comment:
```suggestion
MPAs are different to SPAs in that they have a more standard architecture.
When a new page is loaded it will send a request to the server and all the data will be updated even if it is just a small amount.
A MPA will therefore spend a bit more time loading a new page because it has to update everything.
```